nodebuilder/fraud/constructors.go

```diff
@@ -16,32 +14,46 @@ import (
 	"github.com/celestiaorg/celestia-node/nodebuilder/p2p"
 )
 
-func newFraudService(syncerEnabled bool) func(
-	fx.Lifecycle,
-	*pubsub.PubSub,
-	host.Host,
-	libhead.Store[*header.ExtendedHeader],
-	datastore.Batching,
-	p2p.Network,
-) (Module, fraud.Service, error) {
-	return func(
-		lc fx.Lifecycle,
-		sub *pubsub.PubSub,
-		host host.Host,
-		hstore libhead.Store[*header.ExtendedHeader],
-		ds datastore.Batching,
-		network p2p.Network,
-	) (Module, fraud.Service, error) {
-		getter := func(ctx context.Context, height uint64) (libhead.Header, error) {
-			return hstore.GetByHeight(ctx, height)
-		}
-		pservice := fraudserv.NewProofService(sub, host, getter, ds, syncerEnabled, network.String())
-		lc.Append(fx.Hook{
-			OnStart: pservice.Start,
-			OnStop:  pservice.Stop,
-		})
-		return &Service{
-			Service: pservice,
-		}, pservice, nil
-	}
+func fraudUnmarshaler() fraud.ProofUnmarshaler[*header.ExtendedHeader] {
+	return defaultProofUnmarshaler
+}
+
+func newFraudServiceWithSync(
+	lc fx.Lifecycle,
+	sub *pubsub.PubSub,
+	host host.Host,
+	hstore libhead.Store[*header.ExtendedHeader],
+	registry fraud.ProofUnmarshaler[*header.ExtendedHeader],
+	ds datastore.Batching,
+	network p2p.Network,
+) (Module, fraud.Service[*header.ExtendedHeader], error) {
+	syncerEnabled := true
```

Comment:
do not need this variable anymore as we now have two separate functions

Reply:
it's local to the function, it's too much of a nit to address